graal/src/main/java/io/micronaut/graal/reflect/GraalTypeElementVisitor.java

```diff
@@ -122,11 +122,7 @@ public void visitClass(ClassElement element, VisitorContext context) {
                 packages.add(element.getPackageName());
                 final String beanName = element.getName();
                 addBean(beanName);
-                if (element.isArray()) {
-                    resolveClassData(beanName + "[]");
-                } else {
-                    resolveClassData(beanName);
-                }
+                resolveClassData(beanName + "[]");
```

Comment:
@graemerocher shouldn't this be the other way? I mean remove the one with the array and keep the original?